core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -199,6 +200,22 @@ class CachedDataFrame(df: Dataset[Row], var queryString: String,
     collectInternal().map(boundEnc.fromRow).toArray
   }
 
+  override def withNewExecutionId[T](body: => T): T = queryExecution.executedPlan match {
+    // don't create a new executionId for ExecutePlan since it has already done so
+    case _: ExecutePlan => body
+    case _ =>
```

Comment:
for snc.sql("insert into..." ).show. There are two plans. one is dummy as you said. As discussed, keeping it the way it is because merging these plans is not straightforward.